catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/GetParametersRequestImpl.java

```diff
@@ -88,29 +96,43 @@ public State complete(DAGHolder parameters) throws ProcessingFault, SystemFault
         Result result = getResult(query);
 
         if (result != null) {
-            if (desiredParameters != null) {
-                if (isParamContained(desiredParameters, "ALL")) {
-                    parameters.value = ResultDAGConverter.convertResult(result,
-                            _orb(),
-                            _poa(),
-                            new ArrayList<>());
-                } else if (isParamContained(desiredParameters, "CORE")) {
-                    throw new NO_IMPLEMENT("CORE desired_parameter not supported");
-                } else if (isParamContained(desiredParameters, "ORDER")) {
-                    throw new NO_IMPLEMENT("ORDER desired_parameter not supported");
+            Map<String, List<String>> mandatoryAttributes = new HashMap<>();
+
+            if (outgoingValidationEnabled) {
+                NsiliDataModel nsiliDataModel = new NsiliDataModel();
+                mandatoryAttributes = nsiliDataModel.getRequiredAttrsForView(NsiliConstants.NSIL_ALL_VIEW);
+            }
+            try {
+                if (desiredParameters != null) {
+                    if (isParamContained(desiredParameters, "ALL")) {
+                        parameters.value = ResultDAGConverter.convertResult(result,
+                                _orb(),
+                                _poa(),
+                                new ArrayList<>(),
+                                mandatoryAttributes);
+                    } else if (isParamContained(desiredParameters, "CORE")) {
+                        throw new NO_IMPLEMENT("CORE desired_parameter not supported");
+                    } else if (isParamContained(desiredParameters, "ORDER")) {
+                        throw new NO_IMPLEMENT("ORDER desired_parameter not supported");
+                    } else {
+                        parameters.value = ResultDAGConverter.convertResult(result,
+                                _orb(),
+                                _poa(),
+                                Arrays.asList(desiredParameters),
+                                mandatoryAttributes);
+                    }
                 } else {
-                    parameters.value = ResultDAGConverter.convertResult(result,
-                            _orb(),
-                            _poa(),
-                            Arrays.asList(desiredParameters));
-                }
-            } else {
-                if (result != null) {
-                    parameters.value = ResultDAGConverter.convertResult(result,
-                            _orb(),
-                            _poa(),
-                            new ArrayList<>());
+                    if (result != null) {
```

Comment:
It doesn't look like this check is necessary if you are already checking it on line 98, and won't hit this if there are not desiredParameters.